src/EnergyPlus/PackagedTerminalHeatPump.cc

```diff
@@ -9149,21 +9134,19 @@ void SetOnOffMassFlowRateVSCoil(EnergyPlusData &state,
 void SetMinOATCompressor(EnergyPlusData &state,
                          int const PTUnitNum,                     // index to PTUnit
                          std::string const &cCurrentModuleObject, // type of PTUnit
-                         int const CoolingCoilIndex,              // index of cooling coil
-                         int const HeatingCoilIndex,              // index of heating coil
                          bool &ErrorsFound                        // GetInput logical that errors were found
 )
 {
     bool errFlag = false;
-    auto thisPTUnit = state.dataPTHP->PTUnit(PTUnitNum);
+    auto &thisPTUnit = state.dataPTHP->PTUnit(PTUnitNum);
```

Comment:
Here is the 1 line that needed to be corrected. Unit test fails without this correction. All other changes are clean up.

Reply:
I looked this up. References have the same type signature as values (this is the reason they are evil) and so the compiler cannot infer them. `auto` will *always* be inferred to be a value. If you want the type to be inferred as a reference, you have to write `auto &`. For what it's worth, the same applies to `const`. You have to write `auto const &` if you want the compiler to infer a `const` reference. 

@Myoldmopar, there are currently 2221 uses of `auto` in EnergyPlus that are neither `auto &`, `auto const &`, `auto *`, or `auto const *`. Some of them may still be legitimate, e.g., using `auto` for a function pointer since there is no such thing as a function "value", but most of them should be converted to explicit types (if they are values) or to `auto &`/`auto const &`.